modules/addons/src/gltf/gltf-scenegraph-loader.js

```diff
@@ -10,7 +15,27 @@ async function parse(data, options, uri, loader) {
     DracoDecoder
   });
 
-  return Object.assign({gltfParser, gltf}, createGLTFObjects(options.gl, gltf, options));
+  const gltfObjects = createGLTFObjects(options.gl, gltf, options);
+
+  if (options.waitForFullLoad) {
```

Comment:
This should probably be pulled into a separate function similar to `Query.createPoll`: https://github.com/uber/luma.gl/blob/master/modules/webgl/src/classes/query.js#L133

Reply:
```
async function waitWhileCondition(condition) {
  while (condition()) {
    await new Promise(resolve => window.requestAnimationFrame(resolve));
  }
}
```

Reply:
Does this work?